third_party/web_locale_keymap/lib/web_locale_keymap/key_mappings.g.dart

```diff
@@ -167,22 +168,20 @@ class _StringStream {
 
 Map<String, int> _unmarshallCodeMap(_StringStream stream) {
   final int entryNum = stream.readIntAsVerbatim();
-  return Map<String, int>.fromEntries((() sync* {
-    for (int entryIndex = 0; entryIndex < entryNum; entryIndex += 1) {
-      yield MapEntry<String, int>(stream.readEventKey(), stream.readIntAsChar());
-    }
-  })());
+  return Map<String, int>.fromEntries(
+    Iterable<MapEntry<String, int>>.generate(entryNum, (final int index) =>
+      MapEntry<String, int>(stream.readEventKey(), stream.readIntAsChar())
+  ));
```

Comment:
if we're looking for performance, this would probably be even faster (you'd have to benchmark it to be sure) if we used a regular c-style `for` loop.

Reply:
You could/should use collection literal syntax, here!

```dart
return <String, int>{
  for (var i = 0; i < entryNum; i++)
    stream.readEventKey() : stream.readIntAsChar(),
};
```

Reply:
I never thought it could be done this clean!